examples/bigquery-snowflake-tables-migration-utility/BQ_Table_Creator/bq_table_creator.py

```diff
@@ -0,0 +1,418 @@
+            bigquery.SchemaField("ObjectName", "STRING"),
+            bigquery.SchemaField("Severity", "STRING"),
+            bigquery.SchemaField("Category", "STRING"),
+            bigquery.SchemaField("Message", "STRING"),
+            bigquery.SchemaField("ScriptContext", "STRING"),
+            bigquery.SchemaField("Action", "STRING"),
+        ],
+        skip_leading_rows=1,
+        # The source format defaults to CSV, so the line below is optional.
+        source_format=bigquery.SourceFormat.CSV,
+        write_disposition=bigquery.WriteDisposition.WRITE_TRUNCATE,
+        field_delimiter=",",
+        ignore_unknown_values=True,
+        allow_quoted_newlines=True,
+    )
+    uri = f"gs://{target_bucket_name}/{log_table_id}"
+    try:
+        load_job = client.load_table_from_uri(uri, table_id, job_config=job_config)
+        # Make an API request.
+        load_job.result()  # Waits for the job to complete.
+        destination_table = client.get_table(table_id)  # Make an API request.
+        print(f"Loaded {destination_table.num_rows} rows.")
+    except Exception as ex:
+        print(f"Found the error while creating the table : {str(ex)}")
+
+
+def file_exist(gcs_client, target_bucket_name, log_table_id) -> bool:
+    """
+    Function to check file exist in gcs bucket
+    """
+    bucket = gcs_client.bucket(target_bucket_name)
+    return storage.Blob(bucket=bucket, name=log_table_id).exists(gcs_client)
+
+
+def get_cluster_partitioncolumns(
+    gcs_client,
+    config_source_bucket_name,
+    config_source_prefix,
+    table_name,
+    audit_col_str,
+):
+    """
+    Function to create partion and cluster by column string
+    """
+    config_string = readgcs_file(
+        gcs_client, config_source_bucket_name, config_source_prefix
+    )
+    migration_config_dict = parse_json(config_string)
+    cluster_partition_query = ""
+    for input_val in migration_config_dict["table_config"]:
+        if input_val["table_name"] == table_name:
+            cluster_partition_query = f"{audit_col_str}) \
+                PARTITION BY DATE({input_val['partitioning_field']}) \
+                CLUSTER BY {input_val['clustering_fields']};"
+            return cluster_partition_query
+        else:
+            pass
+
+
+def readgcs_file(gcs_client, bucket_name, prefix):
+    """
+    Function to read gcs file
+    """
+    bucket = gcs_client.get_bucket(bucket_name)
+    blob = bucket.get_blob(prefix)
+    contents = blob.download_as_string()
+    return contents.decode("utf-8")
+
+
+# Parse gcs json file to dictionary
+def parse_json(raw_string):
+    """
+    Function to parse json string
+    """
+    config_parsed_data = ast.literal_eval(raw_string)
+    config_intermediate_data = json.dumps(config_parsed_data)
+    parsed_json_data = json.loads(config_intermediate_data)
+    return parsed_json_data
+
+
+# Write to Blob
+def write_to_blob(gcs_client, bucket_name, file_name, content):
```

Comment:
This function is also defined in `snowflake_ddl_extraction.py`. Could this function be extracted to a common py file to be leveraged by both script without duplicating code?